### PR TITLE
Change the way cutoff for rewards is calculated

### LINKED ISSUE #64: “Oops! Something went wrong” page is shown when user tap on earn Rewards button from hamburger menu
**Frequency:** 100%
**Repro on build version:** Android Internal Build V1.11.0(1004294337), Test Flight Build V1.11.0(47)
**Does not Repro on build version:** Playstore build V1.10.1, App store build V1.10.1,
**Repro on devices:** iPhone 6 + (12.4.5), Samsung galaxy J7 pro (8), iPhone 6s (14.2), iPhone XR (14.4)
Testing Account:   (715) 575 8688 / [Backup Key](https://docs.google.com/spreadsheets/d/1Iy41GGA6e2pES2XHjBn7nb8X3n8zZsu7EW1jvwD2e70/edit#gid=0&range=B12) 
  **Repro Steps:**

1. Launch the application & login with test account
2. Tap on Hamburger menu 
3. Tap on Earn rewards button 
4. Observe   

**Current Behavior:** Oops! something went wrong page is shown when user tap on Eran rewards button .
**Expected Behavior:** User should redirect to Earn rewards page after tapping on Earn rewards page .
**Observation:** App get close when user tap on Quit button on Oops something went wrong page 

- Same error page is shown when user tap on How it works" button from You're eligible CTA.

**Impact:** Users get blocked to earn rewards.
**Attachment:**  [EarnRewrads.MP4](https://drive.google.com/file/d/1dJd9z9LwdHsizZFi2S6rRqxE0qpu8w3Z/view?usp=sharing)

### PR DESCRIPTION
**Pull request by gnardini**
### Description

In this version, we had added a `nextDistribution` field to Firebase. For some reason, it was under the `content` item, which made no sense and caused the app to crash in older versions because it was expecting a string before.

This `nextDistribution` was the date when the next CELO rewards distribution would be made. This is always the following Monday, since that's the cutoff date, so instead of having that value on Firebase now we are just calculating it in the client directly. This will make sure it's always up to date and we don't forget to update it as well.

### Tested

Manually.

### Related issues

- Fixes #64

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/mobile/src/utils/time.ts [hunk 1]
```diff
@@ -341,3 +342,12 @@ function quickFormat(timestamp: number, i18next: i18nType, formatRule: string) {
...
 }
+
+function toUtcDate(date: Date) {
+  return new Date(date.getTime() - date.getTimezoneOffset() * 60 * 1000)
+}
+
+export function getNextMondayAt8pmUtc() {
+  const mondayDate = startOfWeek(Date.now(), { weekStartsOn: 1 })
+  return toUtcDate(mondayDate).getTime() + 20 * ONE_HOUR_IN_MILLIS + 7 * ONE_DAY_IN_MILLIS
+}
```
Comment (jeanregisser) [inline packages/mobile/src/utils/time.ts:353]: This means we're keeping weekly rewards?
I thought we were switching to monthly rewards after March 1st.

File: packages/mobile/src/utils/time.ts [hunk 2]
```diff
@@ -341,3 +342,12 @@ function quickFormat(timestamp: number, i18next: i18nType, formatRule: string) {
     locale: locales[i18next?.language]?.dateFns ?? locales['en-US']?.dateFns,
   })
 }
+
+function toUtcDate(date: Date) {
+  return new Date(date.getTime() - date.getTimezoneOffset() * 60 * 1000)
+}
+
+export function getNextMondayAt8pmUtc() {
+  const mondayDate = startOfWeek(Date.now(), { weekStartsOn: 1 })
```
Comment (jeanregisser) [inline packages/mobile/src/utils/time.ts:351]: Sure this actually returns next Monday?

From the documentation example:
```js
// If the week starts on Monday, the start of the week for 2 September 2014 11:55:00:
var result = startOfWeek(new Date(2014, 8, 2, 11, 55, 0), { weekStartsOn: 1 })
//=> Mon Sep 01 2014 00:00:00
```
https://date-fns.org/v2.18.0/docs/startOfWeek

**Review by jeanregisser — COMMENTED** [review]
It looks like we might want some remote configurability with this.
What do you think?

File: packages/mobile/src/utils/time.ts (hunk 1 above)
Comment (gnardini) [reply packages/mobile/src/utils/time.ts:353]: Yes, we decided to keep weekly rewards so people keep receiving notifications about the program every week and avoid confusion about needing the average balance across a whole month. We will just reduce rewards to be 1/4th of current ones.

File: packages/mobile/src/utils/time.ts (hunk 2 above)
Comment (gnardini) [reply packages/mobile/src/utils/time.ts:351]: This is the previous Monday at 0:00 GMT and we're adding 7 days and 20 hours in the following line, right?

Comment (jeanregisser) [reply packages/mobile/src/utils/time.ts:351]: yes my bad, all good 👍

**Review by jeanregisser — APPROVED** [review]

**Comment by Celoqa** [comment]
Hi @gnardini We have verified this task using  latest Android Internal Build V1.12.0(1004294340) & Test flight build V1.12.0(49) and found mentioned issue is fixed.

**Fixes** #64 

Can you please let us know if we need to test anything else in this task for now.

**Pull request merged**